### PR TITLE
[vividus] Introduce substringBefore/After expressions

### PR DESCRIPTION
**Pull request by boddissattva**
Closes https://github.com/vividus-framework/vividus/issues/3118

### CODE REVIEW

File: vividus/src/test/java/org/vividus/expression/StringsExpressionProcessorTests.java [hunk 1]
```diff
@@ -85,7 +85,9 @@ static Stream<Arguments> expressionSource()
                 arguments("toBase64Gzip(vividus)",                                         "H4sIAAAAAAAA/yvLLMtMKS0GANIHCdkHAAAA"),
                 arguments("escapeHTML(M&Ms)",                                              "M&amp;Ms"),
                 arguments("escapeJSON(\"abc\"\n\"xyz\")",                                  "\\\"abc\\\"\\n\\\"xyz\\\""),
-                arguments("quoteRegExp(Customer(Username))",                               "\\QCustomer(Username)\\E")
+                arguments("quoteRegExp(Customer(Username))",                               "\\QCustomer(Username)\\E"),
+                arguments("substringBefore(123, 2)",                                       "1"),
+                arguments("substringAfter(123, 2)",                                        "3")
```
Comment (ikalinin1) [inline vividus/src/test/java/org/vividus/expression/StringsExpressionProcessorTests.java:90]:
```suggestion
                arguments("substringAfter(123, 2)",                                        "3"),
                arguments("substringAfter(Hello\\, boddissattva, Hello\\,)",                "boddissattva")
```

I believe these expressions should also cover similar cases

Comment (boddissattva) [reply vividus/src/test/java/org/vividus/expression/StringsExpressionProcessorTests.java]: Thanks for your review.
I added another test to handle separator's spaces.
The generic method **substring** could be used by any sort of bi string functions. Therefore, do you prefer another name ?

Comment (uarlouski) [reply vividus/src/test/java/org/vividus/expression/StringsExpressionProcessorTests.java]: @boddissattva can you please check failures in CI and fix them?

Comment (boddissattva) [reply vividus/src/test/java/org/vividus/expression/StringsExpressionProcessorTests.java]: Should be ok now.

**Review by uarlouski — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Comment by valfirst** [comment]
@boddissattva thank you!

**Pull request merged**